unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -1052,6 +1052,21 @@ public void TeleportObjectToFloor(ServerAction action)
 
         public void TeleportFull(ServerAction action) {
 			targetTeleport = new Vector3 (action.x, action.y, action.z);
+            float targetHorizon;
+
+            //check to make sure valid horizon angles are passed in
+            if(!validHorizonAngleValues.ContainsKey(action.horizon))
```

Comment:
You should do `action.horizon % 360` here as someone might be doing things programmatically and thus have generated a horizon that is off by a constant multiple of 360 degrees. This means you'll need to update the keys of `validHorizonAngleValues` also.

Reply:
If `forceAction` is `true` then all of these checks should be ignored.